src/core/symbology-ng/qgsstyle.h

```diff
@@ -273,6 +273,9 @@ class CORE_EXPORT QgsStyle : public QObject
     //! Changes ramp's name
     bool renameColorRamp( const QString& oldName, const QString& newName );
 
+    //! Creates a temporary memory database
+    bool createMemoryDB();
```

Comment:
Is this something we want to make public? I'd lean toward making it private and making TestStyle a friend class of QgsStyle.